### PR TITLE
feat(message-core): add ignoreSameWindow boolean option

### PR DESCRIPTION
**Pull request by danielpickett**
Old behavior for AvMessages is that subscription callbacks are not called when a message event is emitted from the same window where the subscription is registered. This PR adds an `options` object as a third parameter to `avMessages.subscribe` where `ignoreSameWindow` can be set to `false`. This option defaults to `true` to match old behavior. 

Example:
```js
import avMessages from '@availity/message-core'

const avMessages = new AvMessages()

const handleEvent = () => { 
  // ...do some work
}

avMessage.subscribe('my-test-event-name', , { ignoreSameWindow: false } )
```
The above example will call `handleEvent` regardless of whether `my-test-event-name` is dispatched from the same window or a different window such as a child iframe's window.

### CODE REVIEW

File: packages/message-core/src/AvMessage.js [hunk 1]
```diff
@@ -68,10 +81,17 @@ class AvMessage {
...
 
-  onMessage(event, data) {
+  onMessage(event, data, metadata) {
+    const { isSameWindow } = metadata;
+
     if (this.subscribers[event]) {
-      for (const fn of this.subscribers[event]) {
-        fn(data);
+      for (const subscriber of this.subscribers[event]) {
+        const ignoreSameWindow = !!subscriber.options?.ignoreSameWindow;
```
Comment (jordan-a-young) [inline packages/message-core/src/AvMessage.js:89]: Since this is the `subscriber` object you created earlier I think it's safe to not use the `!!` or `?`.

File: packages/message-core/src/AvMessage.js [hunk 2]
```diff
@@ -46,16 +47,28 @@ class AvMessage {
...
+  subscribe(event, callback, options) {
     if (!this.subscribers[event]) {
       this.subscribers[event] = [];
     }
-    this.subscribers[event].push(fn);
+
+    this.#lastId += 1;
+    const id = this.#lastId;
+
+    const ignoreSameWindow = options?.ignoreSameWindow === undefined ? true : options.ignoreSameWindow;
```
Comment (jordan-a-young) [inline packages/message-core/src/AvMessage.js:65]: Double check me, but I think we can shorten this to
`const ignoreSameWindow = options?.ignoreSameWindow ?? true`
If the value is undefined or null, then default to true

**Review by jordan-a-young — CHANGES_REQUESTED** [review]
Looks good! Just have 2 syntax comments

File: packages/message-core/src/AvMessage.js (hunk 1 above)
Comment (danielpickett) [reply packages/message-core/src/AvMessage.js]: Good call. Fixed.

File: packages/message-core/src/AvMessage.js (hunk 2 above)
Comment (danielpickett) [reply packages/message-core/src/AvMessage.js:65]: Good call. Change has been made.

Comment (jordan-a-young) [reply packages/message-core/src/AvMessage.js:65]: I think the change for this line got left of the commit

Comment (danielpickett) [reply packages/message-core/src/AvMessage.js]: Well I do declare! Fixed. Or is it? 🎶 _suspenseful music_ 🎶

**Review by jordan-a-young — APPROVED** [review]
